pyvista/plotting/plotting.py

```diff
@@ -3770,7 +3775,7 @@ def __init__(self, off_screen=None, notebook=None, shape=(1, 1),
 
         def on_timer(iren, event_id):
             """Exit application if interactive renderer stops."""
-            if event_id == 'TimerEvent':
+            if event_id == 'TimerEvent' and self.iren._style != "Context":
```

Comment:
I had to add this when the vtkContextStyleInteractor is active, as otherwise the plotting window was closing immediately. Does this make sense?

Reply:
That's fine with me.  This might lead to issues with animation on Windows as it's quite sensitive to `on_timer` and `update`.  I'll test this on my windows box before merging to see if animating with a chart works on Windows.  Would be a good test to have.

Reply:
Ok perfect, thanks a lot for your comments and fixes!
By animation, do you mean something like the chart_overlays example or something else? Because that example works well on my machine (Windows 10).

Reply:
Animation as in your example you posted in the PR overview.  If that works on windows, it will work on Linux.

Really, this is a great PR and will provide a great way to visualize data from within a vtk plot, especially from a data science perspective.  Appreciate your work!

Reply:
Yes the chart_overlays example is based on that example in the overview, so that should be fine.
Glad that you like it! :D

Reply:
Yeah, let me echo what Alex said. I haven't had the time to review PRs for a while (or contribute otherwise, really), but the additions in this PR look awesome! And being able to embed a matplotlib axes is a very very pleasant surprise to me.